### ISSUE TITLE
An infinite loop hangs big-bang

### ISSUE THREAD

**Issue by blerner** [brownplt/pyret-lang#508]
The following program hangs big-bang:
```
import image as I
import world as W

fun omega():
  omega()
end

fun world-scene(w):
  I.empty-scene(50, 50)
end

fun next-world(w):
  omega()
end
  
# Starts the game
fun main():
  W.big-bang(0,
    [list: W.to-draw(world-scene), 
      W.on-tick(next-world)])
end
```
If you run it, and then run `main()`, an empty window appears, as it should.  If you close the window, it closes... but the browser starts getting laggy.  Surely the program should have been stopped by now!  But if you go to the console, and put a breakpoint inside `iter()` at `val = next.fun(next)`...`omega` is still running.  Something seems to have broken the communications that cause the stack to be stopped and aborted permanently.

**Comment by jpolitz** [comment]
What shows up at the repl when it closes?  Anything?  The close should be handled by the combination of

https://github.com/brownplt/code.pyret.org/blob/master/src/web/js/repl-ui.js#L212

and

https://github.com/brownplt/pyret-lang/blob/master/src/js/trove/world-lib.js#L88

which should cause the big-bang to return the current value for the world as the answer for the computation.  If nothing is printing at the REPL, something may be wrong with that communication.

**Comment by jpolitz** [comment]
Oh.  I see what's happening, interesting case.  I believe the runtime needs a new handler on restarter objects that does "stop-all-with-value", because neither "break" nor "return" handles this correctly.

**Comment by blerner** [comment]
It prints `0` (the only world value it has), and shows a prompt.  But if I press pause in the debugger, it keeps going back into the `omega` frames...

**Comment by jpolitz** [comment]
Yeah this line, or the implementation of `resume`, needs to be more sophisticated; nothing is telling things to stop when there are multiple logical "threads", like the different handlers, running.

https://github.com/brownplt/pyret-lang/blob/master/src/js/trove/world.js#L53

**Comment by blerner** [comment]
BTW, we should try to give the user-aborted error a stack trace, so they can see where the program was potentially hanging when they abort it.

**Comment by sorawee** [comment]
My plot lib needs this to be fixed, so I will take it.
